File: 0706-design-hashmap/0706-design-hashmap.java
import java.util.ArrayList;
import java.util.List;

class ListNode {
    int key;
    int value;
    ListNode next;
    public ListNode(int _key, int _val) {
        key=_key;
        value=_val;
    }
}

class MyHashMap {
    int m;
    List<ListNode> h;
    public MyHashMap() {
        m = 1000;
        h = new ArrayList<>();
        for (int i = 0; i < m; i++) {
            h.add(null);
        }
    }
    
    
    public void put(int key, int value) {
        int index = key % m;
        if (h.get(index) == null) {
            h.set(index, new ListNode(key, value));
        } else {
            ListNode cur = h.get(index);
            ListNode pre= null;
            while (cur!=null) {
                if (cur.key == key) {
                    cur.value = value;
                    return;
                }
                pre=cur;
                cur = cur.next;
            }
            pre.next = new ListNode(key, value);
        }
    }
    
    
    
    public int get(int key) {
        int index = key % m;
        ListNode cur = h.get(index);
        if (cur!=null){
        while (cur != null) {
            if (cur.key == key) {
                return cur.value;
            }
            cur = cur.next;
        }}
        return -1;
    }
    
    public void remove(int key) {
        int index = key % m;
        ListNode cur = h.get(index);
        if (cur == null) return;
        else if (cur.key == key) {
            h.set(index, cur.next);
            cur.next=null;
        } else {
            ListNode prev = null;
            while (cur != null) {
                if (cur.key == key) {
                    prev.next = cur.next;
                    break;
                }
                prev = cur;
                cur = cur.next;    
            }
        }
    }
}

File: 0706-design-hashmap/0706-design-hashmap.py
class ListNode:
    
    def __init__(self,key,val,next=None):
        self.pair=[key,val]
        self.next=next
        
        
class MyHashMap:

    def __init__(self):
        self.s=1000
        self.h=[None]*self.s
        

    def put(self, key: int, value: int) -> None:
        idx=key%self.s
        if not self.h[idx]:
            self.h[idx]= ListNode(key,value)
            return
        else:
            cur=self.h[idx]
            prev=None
            while cur:
                if cur.pair[0]==key:
                    cur.pair[1]=value
                    return
                pre=cur
                cur=cur.next
            pre.next=ListNode(key,value)

    def get(self, key: int) -> int:
        idx=key%self.s
        cur= self.h[idx]
        while cur:
            if cur.pair[0]==key:
                return cur.pair[1]
            cur=cur.next
        return -1
    
    def remove(self, key: int) -> None:
        idx=key%self.s
        cur=self.h[idx]
        if cur is None:
            return 
        elif cur.pair[0]==key:
            self.h[idx]=cur.next
        else:
            pre=cur
            cur=cur.next
            while cur:
                if cur.pair[0]==key:
                    pre.next=cur.next
                    break
                pre=cur
                cur=cur.next
            
        
        


# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)
  # def __init__(self):
  #       self.data = [None] * 1000001
  #   def put(self, key: int, val: int) -> None:
  #       self.data[key] = val
  #   def get(self, key: int) -> int:
  #       val = self.data[key]
  #       return val if val != None else -1
  #   def remove(self, key: int) -> None:
  #       self.data[key] = None